addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/internal/ClientSideHandler.java

```diff
@@ -64,81 +78,208 @@ public void handleMessage(HttpMessageHandlerContext ctx, HttpMessage msg) {
         if (ctx.isFromClient()) {
             return;
         }
+        if (firstHrefId == 0 && msg.getHistoryRef() != null) {
+            // Backstop for looping back through the history
+            firstHrefId = msg.getHistoryRef().getHistoryId();
+        }
 
         historyProvider.addAuthMessageToHistory(msg);
 
-        if (!context.isIncluded(msg.getRequestHeader().getURI().toString())) {
-            return;
+        if (!user.getContext().isIncluded(msg.getRequestHeader().getURI().toString())) {
+            String reqBody = msg.getRequestBody().toString();
+            if (isPost(msg)
+                    && authCreds != null
+                    && StringUtils.isNotEmpty(authCreds.getUsername())
+                    && StringUtils.isNotEmpty(authCreds.getPassword())
+                    && reqBody.contains(authCreds.getUsername())
+                    && reqBody.contains(authCreds.getPassword())
+                    && AuthUtils.getSessionManagementDetailsForContext(user.getContext().getId())
+                            != null) {
+                // The app is sending user creds to another site. Assume this is part of the valid
+                // auth flow and add to the context
+                try {
+                    user.getContext()
+                            .addIncludeInContextRegex(SessionStructure.getHostName(msg) + ".*");
```

Comment:
Log at info?

I think users should be aware (or be able to be aware) of this being done.